collect_app/src/main/java/org/odk/collect/android/application/initialization/ApplicationInitializer.java

```diff
@@ -19,70 +19,57 @@
 
-import static org.odk.collect.android.configure.SettingsUtils.getFormUpdateMode;
+import timber.log.Timber;
 
 public class ApplicationInitializer {
 
     private final Application context;
     private final UserAgentProvider userAgentProvider;
-    private final SettingsPreferenceMigrator preferenceMigrator;
     private final PropertyManager propertyManager;
     private final Analytics analytics;
-    private final Settings generalSettings;
-    private final Settings adminSettings;
-    private final Settings metaSettings;
     private final StorageInitializer storageInitializer;
-    private final AppStateProvider appStateProvider;
-    private final ProjectImporter projectImporter;
-    private final CurrentProjectProvider currentProjectProvider;
-
-    public ApplicationInitializer(Application context, UserAgentProvider userAgentProvider, SettingsPreferenceMigrator preferenceMigrator,
-                                  PropertyManager propertyManager, Analytics analytics, StorageInitializer storageInitializer, Settings generalSettings,
-                                  Settings adminSettings, Settings metaSettings, AppStateProvider appStateProvider, ProjectImporter projectImporter, CurrentProjectProvider currentProjectProvider) {
+    private final LaunchState launchState;
+    private final AppUpgrader appUpgrader;
+
+    @Inject
+    public ApplicationInitializer(Application context, UserAgentProvider userAgentProvider,
+                                  PropertyManager propertyManager, Analytics analytics,
+                                  StorageInitializer storageInitializer, LaunchState launchState,
+                                  AppUpgrader appUpgrader) {
         this.context = context;
         this.userAgentProvider = userAgentProvider;
-        this.preferenceMigrator = preferenceMigrator;
         this.propertyManager = propertyManager;
         this.analytics = analytics;
-        this.generalSettings = generalSettings;
-        this.adminSettings = adminSettings;
-        this.metaSettings = metaSettings;
         this.storageInitializer = storageInitializer;
-        this.appStateProvider = appStateProvider;
-        this.projectImporter = projectImporter;
-        this.currentProjectProvider = currentProjectProvider;
+        this.launchState = launchState;
+        this.appUpgrader = appUpgrader;
     }
 
     public void initialize() {
         initializeStorage();
-        initializePreferences();
+        performUpgradeIfNeeded();
         initializeFrameworks();
         initializeLocale();
-        importExistingProjectIfNeeded();
+
+        launchState.appLaunched();
     }
 
-    private void initializeStorage() {
-        storageInitializer.createOdkDirsOnStorage();
+    private void performUpgradeIfNeeded() {
+        if (launchState.isUpgradedFirstLaunch()) {
+            appUpgrader.upgrade();
+        }
     }
 
-    private void initializePreferences() {
-        performMigrations();
```

Comment:
This needs to move into the new upgrade hook. I've created an issue for this: https://github.com/getodk/collect/issues/4604.